src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
                     }
-                })
-                .With<EndpointWriter.FlushAndStop>(flush => Context.Stop(Self))
-                .With<EndpointWriter.StopReading>(stop =>
+
+                    _writer = CreateWriter();
+                    //Resending will be triggered by the incoming GotUid message after the connection finished
+                    GoToActive();
+                }
+                else
                 {
-                    stop.ReplyTo.Tell(new EndpointWriter.StoppedReading(stop.Writer));
-                    Sender.Tell(new EndpointWriter.StoppedReading(stop.Writer));
-                });
+                    GoToIdle();
+                }
+            });
+            Receive<AttemptSysMsgRedelivery>(redelivery => { }); // Ignore
+            Receive<EndpointManager.Send>(send => send.Message is ISystemMessage, send => TryBuffer(send.Copy(NextSeq())));
+            Receive<EndpointManager.Send>(send => Context.System.DeadLetters.Tell(send));
+            Receive<EndpointWriter.FlushAndStop>(flush => Context.Stop(Self));
+            Receive<EndpointWriter.StopReading>(stop =>
+            {
+                stop.ReplyTo.Tell(new EndpointWriter.StoppedReading(stop.Writer));
+                Sender.Tell(new EndpointWriter.StoppedReading(stop.Writer));
+            });
         }
 
-        protected void Idle(object message)
+        protected void IdleBehavior()
         {
-            message.Match()
-                .With<EndpointManager.Send>(send =>
-                {
-                    _writer = CreateWriter();
-                    //Resending will be triggered by the incoming GotUid message after the connection finished
-                    HandleSend(send);
-                    Context.Become(OnReceive);
-                })
-                .With<AttemptSysMsgRedelivery>(sys =>
+            Receive<IsIdle>(idle => Sender.Tell(Idle.Instance));
+            Receive<EndpointManager.Send>(send =>
+            {
+                _writer = CreateWriter();
+                //Resending will be triggered by the incoming GotUid message after the connection finished
+                HandleSend(send);
+                GoToActive();
+            });
+
+            Receive<AttemptSysMsgRedelivery>(sys =>
+            {
+                if (_resendBuffer.Nacked.Any() || _resendBuffer.NonAcked.Any())
                 {
                     _writer = CreateWriter();
                     //Resending will be triggered by the incoming GotUid message after the connection finished
-                    Context.Become(OnReceive);
-                })
-                .With<EndpointWriter.FlushAndStop>(stop => Context.Stop(Self))
-                .With<EndpointWriter.StopReading>(stop => stop.ReplyTo.Tell(new EndpointWriter.StoppedReading(stop.Writer)));
+                    GoToActive();
+                }
+            });
+            Receive<TooLongIdle>(idle =>
```

Comment:
`TooLongIdle` is one of the new message types introduced in this patch - it's the message sent into response to a `RemoteSettings.QuarantineSilentSystemTimeout`. We've been idle for however long that timeout is (the default is 5 days) and no traffic has exchanged over this network. We will shut down this association inside `HandleTooLongIdle`.